File: CSharp/Unit1/02.Operators and Expressions/10.BitAtPosotionIs1/BitAtPosotionIs1.cs
/*
 * Write a boolean expression that returns if the bit at 
 * position p (counting from 0) in a given integer 
 * number v has value of 1. Example: v=5; p=1  false.
*/
using System;
class BitAtPosotionIs1
{
    static void Main()
    {
        int v = int.Parse(Console.ReadLine());
        int p = int.Parse(Console.ReadLine());
        int mask = 1 << p;
        if ((v & mask) != 0)
        {
            Console.WriteLine("The " + p + " bit is 1");
        }
        else
        {
            Console.WriteLine("The " + p + " bit is 0");
        }
        //another way
        //int bit = v& mask
        //bit = bit >> p;
        //Console.WriteLine("The " + p + " bit is " + bit);
    }
}

File: Java/IntroJava/02.Operators and Expressions/src/BitAtPosotionIs1.java
/*
 * Write a boolean expression that returns if the bit at 
 * position p (counting from 0) in a given integer 
 * number v has value of 1. Example: v=5; p=1  false.
*/
import java.util.Scanner;
public class BitAtPosotionIs1 {

	public static void main(String[] args) {
		Scanner input = new Scanner(System.in);
		int v = input.nextInt();
        int p = input.nextInt();
        input.close();
        int mask = 1 << p;
        if ((v & mask) != 0) {
            System.out.println("The " + p + " bit is 1");
        }
        else {
        	System.out.println("The " + p + " bit is 0");
        }
        
        
        //another way
        int bit = v & mask;
        bit = bit >> p;
        System.out.println("The " + p + " bit is " + bit);
	}

}
